Source/Core/VideoBackends/OGL/Render.cpp

```diff
@@ -1467,6 +1467,13 @@ void Renderer::SwapImpl(u32 xfbAddr, u32 fbWidth, u32 fbStride, u32 fbHeight, co
 	OSD::DoCallbacks(OSD::CallbackType::OnFrame);
 	OSD::DrawMessages();
 
+	if (s_SurfaceNeedsChanged.IsSet())
```

Comment:
Why here? So if this flag is set, you will still apply the old size for postprocessing, and swap the surface directly before the swap call. I guess it's better at the top of this function.